src/ucm/util/reloc.c

```diff
@@ -366,11 +365,23 @@ static ucs_status_t ucm_reloc_dl_info_get(const struct dl_phdr_info *phdr_info,
         goto out;
     }
 
+    if (symtab < (void*)dl_info->start) {
+        symtab = (char*)symtab + dlpi_addr;
+    }
```

Comment:
what is this doing?
maybe add a helper function to reduce code duplication?

Reply:
Right, parsing the headers to get the DT_* value doesn't necessarily get you the final VA of that entity if the library wasn't loaded at 0. So we sanity check to see "is this pointer at least beyond the lowest loaded address we found for this image?", and if not, we add the start to compensate for the offset the binary was actually loaded at. Basically a poor man's relocation.

Reply:
but on other architectures the libraries are also not loaded at 0, and this code works.. is there some different with RISC-V here?
In any case, can use UCS_PTR_BYTE_OFFSET macro to add pointer to offset

Reply:
Since this fix was to specifically address `=reloc` mode I would propose that this commit be dropped from this PR and I open another PR specifically about this incorporating some of the feedback and identifying the exact cause of the offset requirement (and why it isn't needed elsewhere.)

Reply:
> but on other architectures the libraries are also not loaded at 0, and this code works.. is there some different with RISC-V here? In any case, can use UCS_PTR_BYTE_OFFSET macro to add pointer to offset

@yosefe I was able to identify the root cause of the address vs offset situation which I documented in #9283 and addressed in #9284 taking in the feedback to use `UCS_PTR_BYTE_OFFSET` and consolidate where the addition of the base address needs to happen.

Reply:
can we remove it from this PR then?